From: Simon McVittie <smcv@debian.org>
Date: Thu, 31 Mar 2016 18:36:04 +0100
Subject: Load g-i bindings from builddir during build-time testing

Previously, the build-time tests would only pass if the g-i bindings to
OSTree were already installed, with a reasonably similar version.

Signed-off-by: Simon McVittie <smcv@debian.org>
Applied-upstream: 2016.5, commit:https://github.com/ostreedev/ostree/commit/07aa8e1c76463e3de0ffc79d0271773ababf05b4
[smcv: backport to 2016.4]
---
 Makefile-tests.am | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile-tests.am b/Makefile-tests.am
index 9eb9ad8..e7c2395 100644
--- a/Makefile-tests.am
+++ b/Makefile-tests.am
@@ -23,7 +23,10 @@ include $(top_srcdir)/buildutil/glib-tap.mk
 # include the builddir in $PATH so we find our just-built ostree
 # binary.
 TESTS_ENVIRONMENT += OT_TESTS_DEBUG=1 \
-	PATH=$$(cd $(top_builddir) && pwd):$${PATH}
+	GI_TYPELIB_PATH=$$(cd $(top_builddir) && pwd) \
+	LD_LIBRARY_PATH=$$(cd $(top_builddir)/.libs && pwd) \
+	PATH=$$(cd $(top_builddir) && pwd):$${PATH} \
+	$(NULL)
 
 test_scripts = \
 	tests/test-basic.sh \
